mlir/include/mlir/Dialect/LLVMIR/LLVMIntrinsicOps.td

```diff
@@ -1074,6 +1074,12 @@ def LLVM_vector_interleave2
         ]>,
         Arguments<(ins LLVM_AnyVector:$vec1, LLVM_AnyVector:$vec2)>;
 
+def LLVM_vector_deinterleave2
+    : LLVM_IntrOp<"vector.deinterleave2",
+        /*overloadedResults=*/[], /*overloadedOperands=*/[0],
+        /*traits=*/[Pure], /*numResults=*/2>,
```

Comment:
```suggestion
        /*traits=*/[Pure], /*numResults=*/1>,
```
nit: I think the number of results should be 1 since the op returns one struct. Also would it make sense to use `LLVM_OneResultIntrOp` as a base class which sets the number of results for you?